dockerfiles/ink-waterfall-ci/Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG VCS_REF=master
+ARG BUILD_DATE
+ARG REGISTRY_PATH=paritytech
+
+FROM ${REGISTRY_PATH}/ink-ci-linux:production
+
+# metadata
+LABEL io.parity.image.authors="devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.title="${REGISTRY_PATH}/ink-waterfall-ci" \
+	io.parity.image.description="Inherits from paritytech/ink-ci-linux." \
+	io.parity.image.source="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/ink-waterfall-ci/Dockerfile" \
+	io.parity.image.documentation="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/ink-waterfall-ci/README.md" \
+	io.parity.image.revision="${VCS_REF}" \
+	io.parity.image.created="${BUILD_DATE}"
+
+WORKDIR /builds
+
+ENV SHELL /bin/bash
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN	set -eux; \
+	apt-get -y update && \
+# The ink! repository is needed for the `examples/`
+	git clone --depth 1 https://github.com/paritytech/ink.git . && \
```

Comment:
it's not so trivial to get into this dir from GitLab CI. It creates a new dir in the runtime that should be empty, so the regular approach would be either using git submodules, or script in the CI config.
Or do you need this right now in the image buildtime? Then consider cleaning it after.

Reply:
I see, yeah I can also do it in the CI image.

Reply:
it's better doing so yeah. Even though it seems an optimization, one can get into the difference between what you have cloned here and an actual state in the repo, so better cloning every time.